languages/cpp/exercises/concept/basics/.docs/introduction.md

```diff
@@ -0,0 +1,63 @@
+C++ is a statically-typed language, which means that everything has a type at compile-time. Creating a name is referred to as _declaring_ a variable, assigning a value to a declared variable is referred to as _defining_ a variable. Declaration and definition can be performed separately or as part of a single statement. Remember that a declared variable doesn't hold a valid value until defined explicitly but can still be used around the code.
+
+```cpp
+int myVar; // Declaration
+myVar = 10; // Definition
+int anotherVar = 20 // Declaration and definition
+```
+
+A variable can be defined either by explicitly specifying its type, or by letting the C++ compiler infer its type based on the assigned value (known as _type inference_). Therefore, the following two variable definitions are equivalent:
+
+```cpp
+int explicitVar = 10; // Explicitly typed
+auto implicitVar = 10; // Implicitly typed
+```
+
+Updating a variable's value is done through the `=` assignment operator. Once defined, a variable's type can never change.
+
+```cpp
+auto count = 1; // Assign initial value
+count = 2;     // Update to new value
+```
+
+C++ is a multi paradigm language language and functions can be defined in a _class_ or outside. The `class` keyword is used to define a class.
+
+```cpp
+class Calculator {
+    // ...
+};
+```
+
+A function within a class is refered to as a _method_. Each method can have zero or more parameters. All parameters must be explicitly typed, there is no type inference for parameters. Similarly the return type must also be made explicit. There are exceptions to the previous two rules but we'll talk about them in the future. Values are returned from functions using the `return` keyword. To allow a method to be called by code in other files, the `public` access modifier must be added.
```

Comment:
My C++ is more than a bit rusty, so excuse me for asking, but if I read this correctly C++ also allow for functions to be defined outside classes. I'm not sure how common that is. If defining functions outside classes is common, you could consider not using a class here and just have the student directly implement the functions. This would save you from having to introduce classes here (see [F#'s `basics` exercise](https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/basics/.docs/introduction.md) for an example). That said, if most of the rest of the track will use classes, it might be best to introduce it here anyway.

Reply:
You can define functions in the global or a named namespace, in this case the class is acting just a namespace since it doesn't hold any state. It would make sense to avoid using `class`es and `namespace`s for now until we introduce them, I'd avoid also introducing the `static` keyword and the distinction between methods and functions.
Nonetheless I think it's a good idea, I'll change the `string` exercise too and remove the use of `namespace`s there in a later PR.